tests/cephfs/cephfs_mirroring/snapdiff_perf_improvements_across_releases.py

```diff
@@ -0,0 +1,783 @@
+                f"Snaps Synced: {incr2_snapshot_k_info['snaps_synced']}"
+            )
+            log_snapshot_info("Kernel Incremental 2", incr2_snapshot_k_info)
+        if incr2_snapshot_f_info:
+            log.info(
+                f"Fuse Snapshot Info - Name: {incr2_snapshot_f_info['snapshot_name']}, "
+                f"Duration: {incr2_snapshot_f_info['sync_duration']}, "
+                f"Time Stamp: {incr2_snapshot_f_info['sync_time_stamp']}, "
+                f"Snaps Synced: {incr2_snapshot_f_info['snaps_synced']}"
+            )
+            log_snapshot_info("Fuse Incremental 2", incr2_snapshot_f_info)
+
+        log.info("Modify few files and take 3rd incremental snapshots")
+        num_files = 1
+        lines_to_remove = 1
+        snap_k_incr3 = "snap_k_inc3"
+        snap_f_incr3 = "snap_f_inc3"
+        incr3_snapshot_k_info = modify_files_and_take_snapshot(
+            source_clients[0],
+            f"{kernel_mounting_dir_1}{subvol1_path}{io_dir}",
+            num_files,
+            lines_to_remove,
+            snap_k_incr3,
+            fs_util_ceph1,
+            kernel_mounting_dir_1,
+            subvol1_path,
+            io_dir,
+            fs_mirroring_utils,
+            cephfs_mirror_node[0],
+            source_fs,
+            fsid,
+            asok_file,
+            filesystem_id,
+            peer_uuid,
+        )
+        incr3_snapshot_f_info = modify_files_and_take_snapshot(
+            source_clients[0],
+            f"{fuse_mounting_dir_1}{subvol2_path}{io_dir}",
+            num_files,
+            lines_to_remove,
+            snap_f_incr3,
+            fs_util_ceph1,
+            fuse_mounting_dir_1,
+            subvol2_path,
+            io_dir,
+            fs_mirroring_utils,
+            cephfs_mirror_node[0],
+            source_fs,
+            fsid,
+            asok_file,
+            filesystem_id,
+            peer_uuid,
+        )
+
+        log.info("Log incremental snapshot info to CSV file")
+        if incr3_snapshot_k_info:
+            log.info(
+                f"Kernel Snapshot Info - Name: {incr3_snapshot_k_info['snapshot_name']}, "
+                f"Duration: {incr3_snapshot_k_info['sync_duration']}, "
+                f"Time Stamp: {incr3_snapshot_k_info['sync_time_stamp']}, "
+                f"Snaps Synced: {incr3_snapshot_k_info['snaps_synced']}"
+            )
+            log_snapshot_info("Kernel Incremental 3", incr3_snapshot_k_info)
+        if incr3_snapshot_f_info:
+            log.info(
+                f"Fuse Snapshot Info - Name: {incr3_snapshot_f_info['snapshot_name']}, "
+                f"Duration: {incr3_snapshot_f_info['sync_duration']}, "
+                f"Time Stamp: {incr3_snapshot_f_info['sync_time_stamp']}, "
+                f"Snaps Synced: {incr3_snapshot_f_info['snaps_synced']}"
+            )
+            log_snapshot_info("Fuse Incremental 3", incr3_snapshot_f_info)
```

Comment:
Hi @hkadam134 are we doing perf metrics comparision before and after upgrade, manually? We have reference to data in csv for this, can the comparison be done in Automation and report as failure if post upgrade performance has degraded? If manual, we may miss out to check the CSV details at cadence. Please let me know if am missing something here.

Reply:
Hi @sumabai  - As Upgrade in a single flow is not working, I am finding it difficult to have the files in same place.
So, I am appending the results to a file for reference or comparing. Maybe once the upgrade issue is resolved, I will have the comparison logic part of the script.

Reply:
@sumabai  - I added a script to validate the performance results and log the details for now. Since SnapDiff is not fully available, I'm not returning errors at this stage.

Can you review the latest merges.

Reply:
Thanks for the consideration @hkadam134 The script is reporting sync duration difference pre and post upgrade.
Validation would also mean deciding the outcome, pass or fail, which is fail as seen in logs, http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-25GGPI/Validate_snapdiff_perf_results_0.log
So, if its known issue, we can mention BZ ID, else, please plan on how to handle open issues for future cadence runs